### PR TITLE
Deploy the Swin Transformer on TensorRT.

### PR DESCRIPTION
**Pull request by AllentDan**
Support Swin Transformer.

### CODE REVIEW

File: mmdeploy/pytorch/ops/pad.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+import torch.onnx.symbolic_helper as sym_help
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+@FUNCTION_REWRITER.register_rewriter(
```
Comment (grimoire) [inline mmdeploy/pytorch/ops/pad.py:8]: Is this modified from the PyTorch source code?  If it is, we better add a comment here.

File: mmdeploy/pytorch/ops/pad.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Copyright (c) OpenMMLab. All rights reserved.
```
Comment (grimoire) [inline mmdeploy/pytorch/ops/pad.py:1]: Add `modify from ....`

File: mmdeploy/codebase/mmdet/models/transformer.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    x = x.view(B, H, W, C)
+
+    x0 = x[:, 0::2, 0::2, :]  # B H/2 W/2 C
+    x1 = x[:, 0::2, 1::2, :]  # B H/2 W/2 C
+    x2 = x[:, 1::2, 0::2, :]  # B H/2 W/2 C
+    x3 = x[:, 1::2, 1::2, :]  # B H/2 W/2 C
+    x = torch.cat([x0, x1, x2, x3], -1)  # B H/2 W/2 4*C
+    x = x.view(B, -1, 4 * C)  # B H/2*W/2 4*C
+    x = x.view(x.shape[0], x.shape[1], 4,
+               -1).permute(0, 1, 3,
```
Comment (grimoire) [inline mmdeploy/codebase/mmdet/models/transformer.py:39]: `contiguous().view(` -> `reshape`

File: mmdeploy/codebase/mmdet/models/transformer.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmdet.models.utils.transformer.PatchMerging.forward',
+    backend='tensorrt')
+def windowmsa__forward__tensorrt(ctx, self, x, input_size):
```
Comment (grimoire) [inline mmdeploy/codebase/mmdet/models/transformer.py:10]: wrong function name.

File: docs/en/03-benchmark/benchmark.md [hunk 5]
```diff
@@ -754,6 +754,29 @@ Users can directly test the performance through [how_to_evaluate_a_model.md](../
     <td align="center">-</td>
     <td align="center">-</td>
   </tr>
+  <tr>
+    <td align="center" rowspan="2"><a href="https://github.com/open-mmlab/mmdetection/blob/master/configs/swin/mask_rcnn_swin-t-p4-w7_fpn_1x_coco.py">Swin-Transformer</a></td>
```
Comment (RunningLeon) [inline docs/en/03-benchmark/benchmark.md:758]: could update https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/04-supported-codebases/mmdet.md as well.

**Comment by RunningLeon** [comment]
failed on torch==1.8.0
```
RuntimeError: Exporting the operator roll to ONNX opset version 11 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.
```

File: docs/en/03-benchmark/benchmark.md (hunk 5 above)
Comment (AllentDan) [reply docs/en/03-benchmark/benchmark.md:758]: Already added.

File: tests/test_codebase/test_mmdet/test_mmdet_models.py [hunk 6]
```diff
@@ -1576,3 +1576,82 @@ def test_reppoints_head_points2bbox(backend_type: Backend, ir_type: str):
         wrapped_model=wrapped_model,
         model_inputs=rewrite_inputs,
         deploy_cfg=deploy_cfg)
+
+
+@pytest.mark.skipif(
+    reason='Only support GPU test', condition=not torch.cuda.is_available())
+@pytest.mark.parametrize('backend_type', [(Backend.TENSORRT)])
+def test_windows_msa(backend_type: Backend):
```
Comment (RunningLeon) [inline tests/test_codebase/test_mmdet/test_mmdet_models.py:1584]: may need to check whether `tensorrt` is installed by `check_backend(backend)`

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by AllentDan** [comment]
#333

**Review by grimoire — APPROVED** [review]

**Review by lvhan028 — APPROVED** [review]

**Comment by manhtd98** [comment]
ONNX speed slower than Pytorch

**Comment by manhtd98** [comment]
@AllentDan i see that ONNX slower 2times compare with pytorch and TensorRT have same speed with pytorch. Did you know why?

**Pull request merged**